addons/l10n_it_edi/models/account_invoice.py

```diff
@@ -162,6 +162,20 @@ def in_eu(partner):
                 if tax.amount == 0.0:
                     tax_map[tax] = tax_map.get(tax, 0.0) + line.price_subtotal
 
+        # Constraints within the edi make local rounding on price included taxes a problem.
+        # To solve this there is a <Arrotondamento> or 'rounding' field, such that:
+        #   taxable base = sum(taxable base for each unit) + Arrotondamento
+        tax_details = self._prepare_edi_tax_details()
+        for _tax_name, tax_dict in tax_details['tax_details'].items():
+            base_amount = tax_dict['base_amount']
+            tax_amount = tax_dict['tax_amount']
+            tax_rate = tax_dict['tax'].amount
+
+            if float_compare(base_amount, float_round(tax_amount * 100 / tax_rate, 2), 2) \
+               and tax_dict['tax'].price_include and tax_dict['tax'].amount_type == 'percent':
```

Comment:
I would prefer to put the price_include / percent condition first.  What happens if the tax_rate is 0? (or do not they appear in that dict?)